File: src/main/java/no/nav/tag/sykefravarsstatistikk/api/enhetsregisteret/Underenhet.java
package no.nav.tag.sykefravarsstatistikk.api.enhetsregisteret;

import lombok.Builder;
import lombok.Data;
import no.nav.tag.sykefravarsstatistikk.api.domene.Orgnr;

@Builder
@Data
public class Underenhet {
    private final Orgnr orgnr;
    private final Orgnr overordnetEnhetOrgnr;
    private final String navn;
    private final Næringskode5Siffer næringskode;
    private final int antallAnsatte;
}

File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/domenemodeller/Underenhet.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller

sealed class Underenhet {
    abstract val orgnr: Orgnr
    abstract val overordnetEnhetOrgnr: Orgnr
    abstract val navn: String
    abstract val antallAnsatte: Int

    data class Næringsdrivende(
        override val orgnr: Orgnr,
        override val overordnetEnhetOrgnr: Orgnr,
        override val navn: String,
        override val næringskode: BedreNæringskode,
        override val antallAnsatte: Int
    ) : Underenhet(), Virksomhet

    data class IkkeNæringsdrivende(
        override val orgnr: Orgnr,
        override val overordnetEnhetOrgnr: Orgnr,
        override val navn: String,
        override val antallAnsatte: Int
    ) : Underenhet()
}